python/lsst/pipe/drivers/constructCalibs.py

```diff
@@ -998,3 +1089,199 @@ def processSingle(self, sensorRef):
+                          np.min(bgModel.getStatsImage().getArray()),
+                          np.max(bgModel.getStatsImage().getArray()))
+            backgrounds[visit] = bgModel
+            scales[visit] = np.median(bgModel.getStatsImage().getArray())
+
+        return mapToMatrix(pool, self.process, ccdIdLists, backgrounds=backgrounds, scales=scales)
+
+    def measureBackground(self, cache, dataId):
+        """!Measure background model for CCD
+
+        This method is executed by the slaves.
+
+        The background models for all CCDs in an exposure will be
+        combined to form a full focal-plane background model.
+
+        @param cache  Process pool cache
+        @param dataId  Data identifier
+        @return Bcakground model
+        """
+        dataRef = getDataRef(cache.butler, dataId)
+        exposure = self.processSingleBackground(dataRef)
+
+        # NAOJ prototype smoothed and then combined the entire image, but it shouldn't be any different
+        # to bin and combine the binned images except that there's fewer pixels to worry about.
+        config = self.config.largeScaleBackground
+        camera = dataRef.get("camera")
+        bgModel = FocalPlaneBackground.fromCamera(config, camera)
+        bgModel.addCcd(exposure)
+        return bgModel
+
+    def processSingleBackground(self, dataRef):
+        """!Process a single CCD for the background
+
+        This method is executed by the slaves.
+
+        Because we're interested in the background, we detect and mask astrophysical
+        sources, and pixels above the noise level.
+
+        @param dataRef  Data reference for CCD.
+        @return processed exposure
+        """
+        if not self.config.clobber and dataRef.datasetExists("postISRCCD"):
+            return dataRef.get("postISRCCD")
+        exposure = CalibTask.processSingle(self, dataRef)
+
+        # Detect sources. Requires us to remove the background; we'll restore it later.
+        bgTemp = self.subtractBackground.run(exposure).background
+        footprints = self.detection.detectFootprints(exposure, sigma=self.config.detectSigma)
+        image = exposure.getMaskedImage()
+        if footprints.background is not None:
+            image += footprints.background.getImageF()
+
+        # Mask high pixels
+        variance = image.getVariance()
+        noise = np.sqrt(np.median(variance.getArray()))
+        isHigh = image.getImage().getArray() > self.config.maskThresh*noise
+        image.getMask().getArray()[isHigh] |= image.getMask().getPlaneBitMask("DETECTED")
+
+        # Restore the background: it's what we want!
+        image += bgTemp.getImage()
+
+        # Set detected/bad pixels to background to ensure they don't corrupt the background
+        maskVal = image.getMask().getPlaneBitMask(self.config.mask)
+        isBad = image.getMask().getArray() & maskVal > 0
+        bgLevel = np.median(image.getImage().getArray()[~isBad])
```

Comment:
Assuming there are no gradients. And why median here and (clipped) mean in other cases?

Reply:
This is a regular image (as opposed to a background model) that we know has a lot of junk (like stars and galaxies) in it, so the median is a bit more robust. It shouldn't matter, because those pixels are masked, but I'm being careful.